dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -3256,83 +3225,98 @@ BitmapFilterPtr Segment::buildMVCCBitmapFilterStableOnly(
         return elapse_ns / 1'000'000.0;
     };
 
-    auto [skipped_ranges, new_pack_filter_results] = DMFilePackFilter::getSkippedRangeAndFilterForBitmapStableOnly(
-        dm_context,
-        dmfiles,
-        pack_filter_results,
-        start_ts);
+    auto [skipped_ranges, new_pack_filter_results]
+        = DMFilePackFilter::getSkippedRangeAndFilter(dm_context, dmfiles, pack_filter_results, start_ts);
     if (skipped_ranges.size() == 1 && skipped_ranges[0].offset == 0
         && skipped_ranges[0].rows == segment_snap->stable->getDMFilesRows())
     {
-        auto elapse_ms = commit_elapse();
         LOG_DEBUG(
             segment_snap->log,
             "buildMVCCBitmapFilterStableOnly all match, total_rows={}, cost={:.3f}ms",
             segment_snap->stable->getDMFilesRows(),
-            elapse_ms);
+            commit_elapse());
```

Comment:
If you put the lambda `commit_elapse()` here. Then the lambda is not executed when the logging level is "INFO". And the `dm_context.scan_context->build_bitmap_time_ns` is not correct.